### PR TITLE
tests: Bluetooth: Add regression test to l2cap/stress

### PR DESCRIPTION
**Pull request by jori-nordic**
43de309b3e72e7784da9557d091a9bd71a46d7dc made a buffer leak. It was not detected by this test.

The leak is fixed in https://github.com/zephyrproject-rtos/zephyr/pull/71504

Add another configuration, where the final buffer is sent without fragments in order to exercise the leaky code path.

### CODE REVIEW

**Review by theob-pro — DISMISSED** [review]

File: tests/bsim/bluetooth/host/l2cap/stress/src/main.c [hunk 1]
```diff
@@ -479,7 +479,7 @@ static void test_central_main(void)
 	}
 	LOG_DBG("All peripherals disconnected.");
 
-	LOG_DBG("Max segment pool usage: %u bufs", max_seg_allocated);
+	LOG_INF("Max segment pool usage: %u bufs", max_seg_allocated);
```
Comment (jhedberg) [inline tests/bsim/bluetooth/host/l2cap/stress/src/main.c:482]: What's the purpose of this change? (I'm not necessarily opposed to it, but it's not clear to me how it relates to the rest of this change)

Comment (jori-nordic) [reply tests/bsim/bluetooth/host/l2cap/stress/src/main.c:482]: It's not super-related. But I had to bump up the segment pool for the new config. 
It's nice to know what amount was actually used. I moved it to _INF as the _DBG log level kinda floods the output (it's more of a _TRACE level in this test).

**Review by theob-pro — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]
